geps/gep-91/index.md

```diff
@@ -1,22 +1,92 @@
 # GEP-91: Client Certificate Validation for TLS terminating at the Gateway Listener
 
 * Issue: [#91](https://github.com/kubernetes-sigs/gateway-api/issues/91)
-* Status: Provisional
+* Status: Implementable 
 
 (See definitions in [GEP Status][/contributing/gep#status].)
 
 ## TLDR
 
 This GEP proposes a way to validate the TLS certificate presented by the downstream client to the server
-(Gateway Listener in this case) during a [TLS Handshake Protocol][], also commonly referred to as mutual TLS (mTLS).
+(Gateway Listener in this case) during a [TLS Handshake Protocol][].
 
 ## Goals
-- Define an API field to specify the CA Certificate within the Gateway Listener configuration that can be used as a trusted anchor to validate the certificates presented by the client.
+- Define an API field to specify the CA Certificate within the Gateway Listener configuration that can be used as a trust anchor to validate the certificates presented by the client. This use case has been been highlighted in the [Gateway API TLS Use Cases][] document under point 7.
 
 ## Non-Goals
 - Define other fields that can be used to verify the client certificate such as the Certificate Hash or Subject Alt Name. 
 
+### API
+
+* Introduce a `clientValidation` field of type `ClientValidationContext` within [GatewayTLSConfig][] that can be used to validate the client initiating the TLS connection
+to the Gateway.
+* Introduce a `caCertificateRefs` field within `ClientValidationContext` that can be used to specify a list of CA Certificates that
+can be used as a trust anchor to validate the certificates presented by the client.
+
+#### GO
+
+```go
+// ClientValidationContext holds configuration that can be used to validate the client intiating the TLS connection
+// to the Gateway.
+// By default, no client specific configuration is validated.
+type ClientValidationContext struct {
+    // CACertificateRefs contains one or more references to
+    // Kubernetes objects that contain TLS certificates of
+    // the Certificate Authorities that can be used to
+    // as a trust anchor to validate the certificates presented by the client.
+    //
+    // A single CACertRef to a Kubernetes ConfigMap kind has "Core" support.
+    // Implementations MAY choose to support attaching multiple CA certificates to
+    // a Listener, but this behavior is implementation-specific.
+    //
+    // Support: Core - An optional single reference to a Kubernetes ConfigMap,
```

Comment:
Should a Secret with ca.crt be core as well? its not uncommon for things like cert-manager etc to write these

Reply:
good point, we plan on adding support for it in Envoy Gateway, 

for now, its very close to the semantics of BackendTrafficPolicy (which defines a ConfigMap as Core)

adding another new `Core` constraint in this PR may slow down the completion of this PR imo, prefer
if its added incrementally. If the community gives a big thumbs up for this, happy to add it to this PR

Reply:
Added Secret support to Core